core/arch/x86/emit_utils.c

```diff
@@ -3155,6 +3155,50 @@ relink_special_ibl_xfer(dcontext_t *dcontext, int index,
     protect_generated_code(code, READONLY);
 }
 
+/*
+RAW(66) RAW(90)
+RAW(67) RAW(90)
+RAW(f2) RAW(90)
+RAW(f3) RAW(90)
+RAW(66) RAW(66) RAW(66) RAW(66) RAW(66) RAW(90)
+RAW(0f) RAW(1f) RAW(00)
+RAW(0f) RAW(1f) RAW(40) RAW(00)
+RAW(0f) RAW(1f) RAW(44) RAW(00) RAW(00)
+RAW(66) RAW(0f) RAW(1f) RAW(44) RAW(00) RAW(00)
+RAW(0f) RAW(1f) RAW(80) RAW(00) RAW(00) RAW(00) RAW(00)
+RAW(0f) RAW(1f) RAW(84) RAW(00) RAW(00) RAW(00) RAW(00) RAW(00)
+RAW(66) RAW(0f) RAW(1f) RAW(84) RAW(00) RAW(00) RAW(00) RAW(00) RAW(00)
+*/
+bool
+fill_with_nops(dr_isa_mode_t isa_mode, byte *addr, size_t size)
+{
+    if (isa_mode == DR_ISA_AMD64) {
```

Comment:
We usually don't do dynamic isa mode checks in x86 for things like this. I think this should be #ifdef X64
However, the method is already there. Maybe we should split it up into fill_with_nops_arm(dr_isa_mode_t isa_mode, byte *addr, size_t size) and fill_with_nops_x86(byte *addr, size_t size).
@derekbruening , any opinion?

Reply:
Not sure what you mean fill_with_nops_arm() -- the ARM and AArch64 versions are already separated out and already exist.  For x86 64 vs 32, we do use ifdef X64 for code that will never apply to 32-bit builds, but for 64-bit builds we do have support for mixing modes, so the isa_mode check here with both sides should be there for a 64-bit build.

Reply:
@derekbruening , ok.

Looking at the Spec. again, the multi-byte nops seem to be valid for 64- and 32-bit mode, actually. I think we do not need _any_ mode check here. Was there a reason why it was added?

Reply:
Frankly I wasn't sure if they would actually work on 32 bit but I just checked and it seems fine, the check can be removed.

Reply:
Yes thank you, the Spec. - same xref see above - explicitly says this.